python-da/data-prep/src/data_prep.py

```diff
@@ -0,0 +1,122 @@
+    #             raise ValueError(f'File {file_path} is empty or not in the correct format')
+    #         return dataframe
+    #     except pd.errors.ParserError as e:
+    #         # Handle CSV parsing errors
+    #         raise ValueError(f'Failed to parse {file_path} as a CSV file: {e}')
+    #     except Exception as e:
+    #         # Catch any other exceptions that may occur
+    #         raise Exception(f'An unexpected error occurred: {e}')
+
+    def fill_missing_values(self, column_name, method='mode'):
+        # Check if a dataframe is loaded or raise error
+        if self.dataframe is None:
+            raise ValueError('Dataframe is not loaded. Provide a file_path to load dataframe.')
+        
+        # Check if the specified column exists in the dataframe or raise error
+        if column_name not in self.dataframe.columns:
+            raise ValueError(f'Column name {column_name} not found in dataframe')
+        
+        # Check if there are any missing values in the specified column or just return df
+        if self.dataframe[column_name].isnull().sum() == 0:
+            print(f'No missing values found in column {column_name}. Returning original dataframe.')
+            return self.dataframe
+        
+        if method not in ['mode', 'mean', 'median', 'interpolate']:
+            raise ValueError(f'Invalid method: {method}. Supported methods are "mode", "mean", "median", "interpolate".')
+
+
+        #Try-excpet block containing different functionalities for specific keywords
+        try:
+            if method == 'mode':
+                fill_value = self.dataframe[column_name].mode()[0]
+            elif method == 'mean':
+                fill_value = self.dataframe[column_name].mean()
+            elif method == 'median':
+                fill_value = self.dataframe[column_name].median()
+            elif method == 'interpolate':
+                self.dataframe[column_name].interpolate(inplace=True)
+                return self.dataframe
+            
+            # Fill missing values in the dataframe
+            self.dataframe[column_name].fillna(fill_value, inplace=True)
+        except Exception as e:
+            raise Exception(f'An error occurred while filling missing values: {e}')
+
+        return self.dataframe
+    
+
+    # Removes duplicate rows from the dataframe. Limited functionality for now, aim to add more 
+    def remove_duplicates(self):
+        
+        #File check
+        if self.dataframe is None:
+            raise ValueError('Dataframe is not loaded. Provide a file_path to load dataframe.')
+
+        # Remove duplicates using pandas drop_duplicates method
+        initial_rows = len(self.dataframe)
+        self.dataframe.drop_duplicates(inplace=True)
+        removed_rows = initial_rows - len(self.dataframe)
+
+        print(f'Removed {removed_rows} duplicate rows.')
+        return self.dataframe
+    
+    def remove_z_score_outliers(self, threshold: float = 3.0) -> None:
```

Comment:
Should we coordinate after the data profiling checks return the outliers ? It's currently operational, and I'll approve it for now. However, we need to connect and understand where the data profiling checks store their results, so you can retrieve them. Alternatively, consider importing the DataQuality class and utilising its functions. Let's discuss the best approach.